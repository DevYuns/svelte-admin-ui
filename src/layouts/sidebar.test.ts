import SideBar from './sidebar.svelte';
import {render} from '@testing-library/svelte';
import {setUpUser} from './../../test/testUtils';

describe('render test', () => {
    it('should render without crash', () => {
        const {container} = render(SideBar, {isMobile: false});

        expect(container).toMatchSnapshot();
    });
});

describe('interaction test', () => {
    it('should change width when a user click icon', async () => {
        const {user, getByRole, getByTestId} = setUpUser(SideBar, {isMobile: false});

        const container = getByTestId('container');
        const icon = getByRole('img');

        const currentWidth = window.getComputedStyle(container).width;

        await user.click(icon);

        const changedWidth = window.getComputedStyle(container).width;

        expect(currentWidth).not.toBe(changedWidth);
    });
});