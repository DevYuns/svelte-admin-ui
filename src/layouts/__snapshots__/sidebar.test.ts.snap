// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test should render without crash 1`] = `
<body>
  <div>
    <aside
      class="svelte-1jcti8y"
      data-testid="container"
    >
      <img
        alt="hamburger"
        class="icon svelte-1jcti8y"
        src="[object Object]"
      />
       
      <div
        class="profile svelte-1jcti8y"
      >
        Profile
      </div>
       
      <nav
        class="svelte-1jcti8y"
      >
        <ul
          class="svelte-1jcti8y"
        >
          <li>
            <a
              class="svelte-1jcti8y active"
              href="mockURL"
            >
              Home
            </a>
          </li>
           
          <li>
            <a
              class="svelte-1jcti8y active"
              href="mockURL"
            >
              User
            </a>
          </li>
        </ul>
      </nav>
    </aside>
  </div>
</body>
`;
